File: algorithm/baekjoon/codeplus/first-solve/src/main/java/com/keunwon/algorithm/greedy/Problem13305.java
package com.keunwon.algorithm.greedy;

import java.util.Scanner;

public class Problem13305 {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int n = sc.nextInt();
        int[] distances = new int[n - 1];
        int[] fees = new int[n];

        for (int i = 0; i < n - 1; i++) {
            distances[i] = sc.nextInt();
        }

        for (int i = 0; i < n; i++) {
            fees[i] = sc.nextInt();
        }

        long amount = 0L;
        long min = Long.MAX_VALUE;
        for (int i = 0; i < n  -1; i++) {
            if (fees[i] < min) {
                min = fees[i];
            }
            amount += min * distances[i];
        }

        System.out.println(amount);
    }
}

File: algorithm/kotlin-algorithm/src/main/kotlin/com/keunwon/algorithm/baekjoon/Problem13305.kt
package com.keunwon.algorithm.baekjoon

/**
 * <p>
 * 이름: 주유소
 * 난이도: 실버-3
 * </p>
 *
 * @see <a href="https://www.acmicpc.net/problem/13305">주유소 (실버-3)</a>
 **/
class Problem13305 {
    fun solution(distances: IntArray, prices: IntArray): Long {
        var result = 0L
        var min = Long.MAX_VALUE

        for ((i, d) in distances.withIndex()) {
            if (min > prices[i]) min = prices[i].toLong()
            result += min * d
        }
        return result
    }
}

fun main() {
    val n = readln().toInt()
    val distances = run {
        val arr = readln().split(" ").map { it.toInt() }
        IntArray(n - 1) { arr[it] }
    }
    val prices = run {
        val arr = readln().split(" ").map { it.toInt() }
        IntArray(n) { arr[it] }
    }

    Problem13305().solution(distances, prices).also { println(it) }
}
